pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -180,9 +180,31 @@ func loadMonConfig(clientset kubernetes.Interface, namespace string) (map[string
 }
 
 func createClusterAccessSecret(clientset kubernetes.Interface, namespace string, clusterInfo *cephconfig.ClusterInfo, ownerRef *metav1.OwnerReference) error {
-	logger.Infof("creating mon secrets for a new cluster")
+	logger.Infof("creating csi and mon secrets for a new cluster")
 	var err error
 
+	// Store the admin secret for the csi driver
+	csiSecrets := map[string][]byte{
+		// adminID is expected for the cephfs driver
+		"adminID":  []byte("admin"),
+		"adminKey": []byte(clusterInfo.AdminSecret),
+		// userID is expected for the cephfs driver
+		"userID":  []byte("admin"),
+		"userKey": []byte(clusterInfo.AdminSecret),
+	}
+	csiSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-ceph-csi",
+			Namespace: namespace,
+		},
+		Data: csiSecrets,
+		Type: k8sutil.RookType,
+	}
+	k8sutil.SetOwnerRef(&csiSecret.ObjectMeta, ownerRef)
+	if _, err = clientset.CoreV1().Secrets(namespace).Create(csiSecret); err != nil {
```

Comment:
@travisn its good to validate the err for `isAlreadyExist`.

Reply:
Yes, nice catch